File: 15.03/consecutiveNumbers.py
#!/usr/bin/python
#/usr/bin/python3.5

'''
Desafio H
Consecutive Numbers Gym - 101807C
q numero de perguntas
para cada q responder se x pode ser representado pela soma de pelo menos dois inteiros positvos consecutivo

A soma de dois números consecutivos é ímpar, pois um deles deve ser par e outro ímpar. 
2 n = 2 n-1 + 2 n-1
'''

def soma(x):
    return ((x & (x-1)) and x)

    
def main():
    q = int(input())
    i = 0


    while(i<q):
        x = int(input())

        res = soma(x)

        if(res == False):
            print("NO")
        else:
            print("YES")      
                    
        i+=1
    

main()

File: 15.03/consecutiveNumbers.c
#include<stdio.h>
#include<stdlib.h>



/*
Desafio H
Consecutive Numbers Gym - 101807C
q numero de perguntas
para cada q responder se x pode ser representado pela soma de pelo menos dois inteiros positvos consecutivo

A soma de dois números consecutivos é ímpar, pois um deles deve ser par e outro ímpar. 
2 n = 2 n-1 + 2 n-1
*/

int consecutive (int number){
    int vetor [number];
    long long int soma;

    for(i=0;i<number/2;i++)
}

int main (){
    int q,i=0, res = 0;
    int long long number;

    scanf("%d",&q);


    while(i<q){
        scanf("%d", number);

        res = consecutive(number);

        if(res == 10)
            printf("yes");
        else
            printf("no");

    }
    return 0;
}